### ISSUE TITLE
RFE: possibility to create resources on top of a file 

### ISSUE THREAD

**Issue by misacek007** [OndrejHome/ansible.pcs-modules-2#8]
All resources/constraints would be created on top of a file (`pcs -f <file>`) which would be in the end pushed to the cluster (or not to do this manually).

**Comment by OndrejHome** [comment]
Hi @misacek007, 
Thank you for suggestion.

Use of alternative file for cluster configuration (CIB) would make sense to all modules except following:
- `pcs_auth` - this doesn't touch cluster CIB in any way
- `pcs_cluster` - this seems to change more than just CIB when used

For the other modules this looks possible on first look. Testing just randomly it looks that sometimes there would be good if there was already some configuration from which this would start and not just empty one - empty configuration doesn't contain any nodes and some commands such as location constraints and surprisingly while creating location constraint for non-existing node works fine, the resource defaults and resource operation defaults fails on my testing system demanding some node name.

As first step I see here only the ability to specify file for `pcs -f <file>`. Maybe it would be interesting to have ability to dump config from existing cluster and then push it whole back when the changes to file are done, but this seems as something that can be done by just using `command` module manually with commands.

If you have any feedback to above feel free to update this issue. I will have a look at this during some weekend.

TODO for implementation:
- [ ] `pcs_resource`
- [ ] `pcs_resource_defaults`
- [ ] `pcs_property`
- [ ] `pcs_constraint_colocation`
- [ ] `pcs_constraint_location`
- [ ] `pcs_constraint_order`
- [ ] check which modules requires existing cluster config to operate properly and which are happy with empty one (hopefully all will be happy with empty one)

**Comment by misacek007** [comment]
Just to summarize that I got it right for transactional approad: you would specify for example `file: /tmp/cluster.cib` in several resources/constraints and do `cib-push` after using `command` module. Right?

The rest sounds good to me.

**Comment by OndrejHome** [comment]
yes, exactly as you write. I will try to get this over upcoming weekend so you can test it out.

**Comment by OndrejHome** [comment]
Hi @misacek007, in the `develop` branch of repo you will find the updated version of modules to use new attribute `cib_file` that can be used to make changes to file (`pcs -f cib_file`) instead of live cluster configuration. All changed modules except `pcs_property` and `pcs resource_defaults` makes a basic check if the provided `cib_file` is a parseable XML and leaves rest of checks in `pcs` later.

Would you be able to try the changed modules and let me know if they work well for your use case or if there are any issues that you have encountered using them. If there is positive feedback or no negative feedback on changes in next week I will go ahead and merge the changes into master and mark them as new version of `OndrejHome.pcs-modules-2`.

**Comment by OndrejHome** [comment]
Changes are now merged in `master` and available with tag `v13` including on Ansible galaxy. If the problem still persist please let me know,  if not I will close this issue this week.

**Comment by misacek007** [comment]
I just wanted to let you know that this changes solved perfectly my needs. Thanks a lot!

**Comment by OndrejHome** [comment]
Thank you for feedback that it works well!
